browser/brave_content_browser_client.cc

```diff
@@ -618,5 +619,13 @@ BraveContentBrowserClient::CreateThrottlesForNavigation(
     throttles.push_back(std::move(ipfs_navigation_throttle));
 #endif
 
+  if (std::unique_ptr<content::NavigationThrottle>
+          domain_blocked_navigation_throttle = brave_shields::
+              DomainBlockedNavigationThrottle::MaybeCreateThrottleFor(
```

Comment:
nit: the naming of this is a bit unclear to me, can we call it FirstPartyBlocking or something along those lines?

Reply:
I'd recommend against `FirstPartyBlocking` since it might get incorrectly conflated with either of:
- network filters with a `$first-party` option
- Brave Shields' option to block first-party cosmetic page resources (aggressive mode)

If this is confusing, I'd suggest something like `TopLevelDocumentBlocking`, `StrictBlocking` (borrowing uBO's terminology), or perhaps even just `DocumentBlocking` since the corresponding filter rules use a `$document` option.

Reply:
I'm not a fan of `StrictBlocking` because it's not very descriptive, but I would also be ok with just dropping the `ed/ing` and calling this `DomainBlock` which is consistent with `AdBlock`

Reply:
that is assuming that we're blocking the domain in 1p and 3p contexts

Reply:
Renamed everything to `DomainBlock`